authors/apps/authentication/serializers.py

```diff
@@ -143,3 +144,30 @@ def update(self, instance, validated_data):
         instance.save()
 
         return instance
+
+
+
+class PasswordResetSerializer(serializers.Serializer):
+    email = serializers.EmailField(required=False)
+    message = serializers.EmailField(max_length=255, required=False)
+    token = serializers.EmailField(max_length=255, required=False)
+
+    class Meta:
+        model = User
+        fields = ('token')
+
+    def validate(self, data):
+        email = data.get('email', None)
+        if email is None:
+            raise serializers.ValidationError(
+                'An email is required to reset password.'
+            )
+        user = get_object_or_404(User, email=email)
+        # user.is_active=False
+        
+        # return token here
+        return {
+            'email': user,
+            'token':user.token(0.0208333),
```

Comment:
Does this token override the JWT token payload that is generated in the models?

Reply:
the method to create a token in the models allows you to change the default value of the number of days the token is valid.  So by turning 30 minutes into days, I can change how long the token generated is valid